libs/barista-components/bar-indicator/src/_bar-indicator-theme.scss

```diff
@@ -1,20 +1,17 @@
-@import '../../core/src/theming/theming';
-
-@mixin dt-theme-bar-indicator($theme) {
-  $palette-names: 'main', 'recovered', 'error';
-  $default-palette: 'main';
-
-  @each $name in $palette-names {
-    $palette: dt-get-theme-palette($theme, $name);
-    $default-color: dt-get-theme-color($palette, 'default');
+:host.dt-color-main {
+  .dt-bar-indicator-bar {
+    background-color: var(--dt-bar-indicator-default-color);
+  }
+}
```

Comment:
I'd suggest to restructure this file a bit. 


1. define a one css custom property for the dt-bar-indicator-background depending on the color set on the host
2. create one css class that sets the value of the background to the css custom property. 
```
.host.dt-color-main {
  --dt-bar-indicator-background-color: var(--dt-theme-main-default-color);
}

... the other colors the same way

:host .dt-bar-indicator-bar {
  background: var(--dt-bar-indicator-background-color);
}
```

Think about custom properties like having a placeholder in the standard selector that gets a value at runtime in the browser inherited from some parent.

Reply:
Is there a way to get the royalblue `--dt-theme-main-default-color` property for the main bar indicator, even if I'm not in its theme?
Changing the styles as suggested will change the main color to turquoise.